### ISSUE TITLE
Bug: calcite-alert with yellow color doesn't meet WCAG color contrast requirements with light theme 

### ISSUE THREAD

**Issue by kellyhutchins** [Esri/calcite-components#994]
### Summary

The calcite-alert provides 4 options for alert colors (blue, green, red and yellow). The yellow option doesn't meet color contrast requirements. 
https://webaim.org/resources/contrastchecker/?fcolor=EDD317&bcolor=FFFFFF

**Comment by macandcheese** [comment]
cc @bstifle re: our yellow color.

@kellyhutchins - does the accent color need to pass? The text content itself should really convey the severity / status of the message and we won't be able to tell what the color of whatever element the alert is positioned over is.

I tried to look for other examples - IBM's carbon "warning" yellow similarly doesn't pass on its own, but in the demo it's presented on a dark background (https://www.carbondesignsystem.com/components/notification/usage/), which seems unrealistic since the alert would "floating" be over any number of page elements.

**Comment by kellyhutchins** [comment]
The accent color doesn't need to pass on its own since as you noted the text should convey the warning.  But visually impaired users will have difficulty seeing the warning icon but not the other alert icons (error, light bulb etc). 

Reading the spec it says that meaningful graphics need to meet the contrast requirements. I think the warning icon on the alert component is meaningful but if the message text is well defined and notes that its a warning then I guess the icon can be considered supplementary. 

https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

**Comment by macandcheese** [comment]
OK, yeah that's fair, we don't necessarily want to have dev users need to prepend the text with "warning: <message>", so maybe we can just look at updating that color. There are a few other places where we have a "color-variable-*accessible*" variant so maybe we could use something like that here 👍

**Comment by kellyhutchins** [comment]
@macandcheese  I took a look at the recommendation from the company doing a11y testing for config apps and they recommend providing a screen reader only text equivalent.  So based on their notes it sounds like the yellow is ok as long as we add the text equivalent. 

Here's the recommendation from their report: 

> Text equivalents should be concise and meaningful replacements for images. The yellow warning icons should have text associated with them using visually hidden text.

```
<svg role="img" focusable="no" aria-label="Warning" class="svg"></svg>
<span class="sr-only">Warning</span>

.sr-only{
   position: absolute !important;
  height: 1px; 
  width: 1px; 
  overflow: hidden;
  padding: 0 !important;
  border: 0! important;
  white-space: nowrap !important;
  clip: rect(1px 1px 1px 1px) !important; /* IE6, IE7 */
  clip: rect(1px, 1px, 1px, 1px) !important;
  clip-path: inset(50%) !important;
}
```

**Comment by julio8a** [comment]
This issue is still valid.

**Comment by bstifle** [comment]
agreed. there's no real way to make yellow 3:1 on white backgrounds. The text is primary for alerts, so the icon and bar are more flourish and supportive

**Comment by julio8a** [comment]
@bstifle, this is something we should document in the alert page on the doc site. The yellow is more decorative and users should not rely on it to convey their message.

**Comment by macandcheese** [comment]
@bstifle @julio8a - not sure if this is still actionable, feel free to follow up if so.

**Comment by bstifle** [comment]
The way i see it, we either remove yellow, or go with this gold-brackish color:

![image](https://user-images.githubusercontent.com/14083530/118529138-c81e9180-b708-11eb-8d9c-460c32897ba9.png)

![image](https://user-images.githubusercontent.com/14083530/118529318-00be6b00-b709-11eb-8d4a-9c81f3023d32.png)


keep in mind that only hits 3:1 on white. It would fail on f8f8f8, f3f3f3. and eaeaea.

**Comment by julio8a** [comment]
I don't think there's anything actionable here. We don't want to change it as this is more of a decorative element and should not be relied on for notifying the user.

**Comment by julio8a** [comment]
Closing this. Please reopen if still valid.
